samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/samples/FileTransferIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+                .when()
+                        .post(FILE_TRANSFER_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().asString();
+
+        // Verify TransferProcessId
+        assertThat(transferProcessId).isNotNull();
+
+        // Verify file transfer is completed and file contents
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                assertThatJson(fetchTransfer(transferProcessId).toString()).and(
+                        json -> json.node("id")
+                                .withFailMessage("TransferProcessId not matched")
+                                .isEqualTo(transferProcessId),
+                        json -> json.node("state")
+                                .withFailMessage("TransferProcess is not in COMPLETED state")
+                                .isEqualTo(TransferProcessStates.COMPLETED.code())
+                ));
+
+        var copiedFilePath = Path.of(format(CONSUMER_ASSET_PATH + "/%s.txt", PROVIDER_ASSET_NAME));
+        var actualFileContent = fetchFileContent(copiedFilePath);
+        assertThat(actualFileContent)
+                .withFailMessage("Transferred file contents are null")
+                .isNotNull();
+        assertThat(actualFileContent)
+                .withFailMessage("Transferred file contents are not same as the source file")
+                .isEqualTo(fileContent);
+    }
+
+    private ObjectNode fetchTransfer(String transferProcessId) {
+        return
+                given()
+                        .pathParam(TRANSFER_ID_PARAM, transferProcessId)
+                .when()
+                        .get(TRANSFER_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().as(ObjectNode.class);
+    }
+
+    /**
+     * Fetch negotiated contract agreement.
+     *
+     * @param contractNegotiationRequestId ID of the ongoing contract negotiation between consumer and provider.
+     * @return Negotiation as {@link ObjectNode}.
+     */
+    private ObjectNode fetchNegotiatedAgreement(String contractNegotiationRequestId) {
+        return
+                given()
+                        .pathParam(CONTRACT_NEGOTIATION_REQUEST_ID_PARAM, contractNegotiationRequestId)
+                        .header(API_KEY_HEADER, API_KEY_CONTROL_AUTH)
+                .when()
+                        .get(CONTRACT_AGREEMENT_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().as(ObjectNode.class);
+    }
+
+    /**
+     * Helper method to read file contents on the given {@link Path}
+     *
+     * @param filePath see {@link Path}
+     * @return Contents of file as a {@link String} or null if file does not exist.
+     */
+    private String fetchFileContent(Path filePath) {
+        if (filePath.toFile().exists()) {
```

Comment:
this if is not needed since the "file doesn't exist" case is already handled by the try catch block